### PR TITLE
20 instagram integration

### LINKED ISSUE #20: Instagram api integration
As a first huge step we need to get images (any kind) from instagram. this is the first must have step. to do such task our app needs integration with instagram API (Application programming interface).

https://www.instagram.com/developer/

С#

https://github.com/InstaSharp/InstaSharp

https://github.com/a-legotin/InstaSharper

### PR DESCRIPTION
**Pull request by Dicrit**
closes #20 
Good way to start. Controller is needed to be reworked in future when authorization will be implemented.

### CODE REVIEW

File: InstaBotPrototype/InstaBotLibrary/Instagram/InstagramUnauthorizedException.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace InstaBotLibrary.Instagram
+{
+    class InstagramUnauthorizedException : InvalidOperationException
+    {
+        public InstagramUnauthorizedException() : base("Instagram wasn't auntificated, please use Auth method before")
```
Comment (msvaillant) [inline InstaBotPrototype/InstaBotLibrary/Instagram/InstagramUnauthorizedException.cs]: typo - authentification

File: InstaBotPrototype/InstaBotPrototype/InstaBotPrototype.csproj [hunk 2]
```diff
@@ -9,6 +9,7 @@
   </ItemGroup>
   
   <ItemGroup>
+    <PackageReference Include="InstaSharp" Version="2.0.6" />
```
Comment (msvaillant) [inline InstaBotPrototype/InstaBotPrototype/InstaBotPrototype.csproj]: I think is not needed in Web part

File: InstaBotPrototype/InstaBotPrototype/Instagram/Controllers/InstagramController.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+namespace InstaBotPrototype.Instagram
+{
+    [Route("[controller]/[action]")]
+    public class InstagramController : Controller
+    {
+        InstagramService instagramService;
+
+        public InstagramController(IConfiguration configuration)
+        {
+            instagramService = new InstagramService(configuration);
```
Comment (msvaillant) [inline InstaBotPrototype/InstaBotPrototype/Instagram/Controllers/InstagramController.cs]: Inject and register service first, don't use NEW keyword, use ASP.Net Core DI instead

File: InstaBotPrototype/InstaBotLibrary/Instagram/InstagramService.cs [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+{
+    public class InstagramService : IInstagramService
+    {
+        private InstagramConfig instagramConfig;
+        private OAuthResponse auth = null;
+        public InstagramService(InstagramConfig config)
+        {
+            instagramConfig = config;
+        }
+        public InstagramService(IConfiguration configuration)
```
Comment (msvaillant) [inline InstaBotPrototype/InstaBotLibrary/Instagram/InstagramService.cs]: read this article first
https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/options

**Review by msvaillant — CHANGES_REQUESTED** [review]

File: InstaBotPrototype/InstaBotPrototype/InstaBotPrototype.csproj (hunk 2 above)
Comment (Dicrit) [reply InstaBotPrototype/InstaBotPrototype/InstaBotPrototype.csproj]: Completely agree. It left from time when all logic was on the main project.

**Review by msvaillant — APPROVED** [review]

**Pull request merged**